### DISCUSSION TITLE
Change the delimiter for Ctrl+backspace in a terminal

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#4588] [Q&A]
Consider this text, with the cursor `|` positioned at the end:

    # rm myfile.txt|

When working in an editor in code-server (3.12.0, [Ubuntu-based](https://github.com/cdr/code-server/discussions/4529#discussioncomment-1717543)), pressing Ctrl-backspace when the cursor is at the end will remove just `txt` to get back to the `.`:

    # rm myfile.|

In other words, the delimiter for this action is a `.` or a space.

But when working in a terminal in code-server, the same action will *instead* delete right back to the next space, i.e. treating `.` like any other character:

    # rm |

In other words, the delimiter for this action is just a space, not a `.` .

How do I get the former behaviour throughout (i.e. word delimiter is `.` or space), even in a terminal?  The former behaviour is what I'm used to... seems pretty common, e.g. even in this GitHub editor panel.

**Comment by jsjoeio** [comment]
Hmm...this is an interesting problem. Does the same behavior happen in VS Code, Codespaces, https://vscode.dev/, and https://xtermjs.org/?

I would have to assume this is upstream but we couldn't arrive at that conclusion without testing other environments to see if the same behavior exists. It could be a difference between monaco-editor and xtermjs and which delimiters they use.

**Reply by drmrbrewer** [reply]
I haven't tried a local copy of VS Code or in Codespaces, but I suspect that that behaviour will be the same.. it seems to be almost universal that the delimiter for a Ctrl+backspace operation is anything that isn't a letter or number.  Certainly it's the same in the editor on vscode.dev.  The Ctrl+backspace shortcut doesn't work at all on the terminal on xtermjs.org (it just deletes back one character).  The only time I've seen Ctrl+backspace treat `.` as a letter is in the terminal on code-server.  Maybe others like it that way, but I'm not sure why... so many parts of the programming world use `.` to delimit words or units that it seems odd to blast through them when doing a Ctrl+backspace.  It's not even consistent with navigating left/right in the terminal whilst holding down Ctrl... doing a Ctrl+left_arrow with the cursor `|` on `myfile.json|` in the terminal moves back to `myfile.|json` as I'd expect... but Ctrl+backspace from the same position will just wipe out the whole thing.

**Reply by jsjoeio** [reply]
> I haven't tried a local copy of VS Code or in Codespaces, but I suspect that that behaviour will be the same..

Ah got it! In that case, might be worth raising an issue upstream.

> Certainly it's the same in the editor on vscode.dev.

Right! Though there isn't a terminal. Just trying to see if this is specific to the browser by chance. 

> The only time I've seen Ctrl+backspace treat . as a letter is in the terminal on code-server.

Interesting! I wonder then if it reproduces in Codespaces? I would be very very surprised if we had changed this behavior ourselves. That seems unlikely.

Well if you find time to reproduce in VS Code or Codespaces, let me know! I'm curious
